scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -578,7 +587,11 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
             sepRegions.head.isInstanceOf[RegionCase] ||
             sepRegions.head.isInstanceOf[RegionEnum] ||
             sepRegions.head.isInstanceOf[RegionIndent] ||
-            sepRegions.head.isInstanceOf[RegionIndentEnum])
+            sepRegions.head.isInstanceOf[RegionIndentEnum] ||
```

Comment:
i think it might be time to change this entire parenthesized expression `(sepRegions.isEmpty || ...)` to
```
sepRegions.headOption.forall {
  case _: RegionBrace | _: RegionCase | _: RegionEnum => true
  case _: RegionIndent | _: RegionIndentEnum => true
  case x: RegionParen => x.canProduceLF
  case _ => false
}
```